Gems/Archive/Code/Source/Clients/ArchiveReader.cpp

```diff
@@ -0,0 +1,928 @@
+                    listResult.m_filePathToken = static_cast<ArchiveFileToken>(filePathIndex);
+
+                    // Gather the file metadata
+                    const ArchiveTocFileMetadata& fileMetadata = m_archiveToc.m_tocView.m_fileMetadataTable[filePathIndex];
+
+                    // Use the compression algorithm index to lookup the compression algorithm ID
+                    if (fileMetadata.m_compressionAlgoIndex < m_archiveHeader.m_compressionAlgorithmsIds.size())
+                    {
+                        listResult.m_compressionAlgorithm = m_archiveHeader.m_compressionAlgorithmsIds[
+                            fileMetadata.m_compressionAlgoIndex];
+                    }
+
+                    listResult.m_uncompressedSize = fileMetadata.m_uncompressedSize;
+                    if (auto rawFileSizeResult = GetRawFileSize(fileMetadata, m_archiveToc.m_tocView.m_blockOffsetTable);
+                        !rawFileSizeResult)
+                    {
+                        fileResultOutcome = AZStd::unexpected(AZStd::move(rawFileSizeResult.error()));
+                        return;
+                    }
+                    else
+                    {
+                        listResult.m_compressedSize = rawFileSizeResult.value();
+                    }
+                    listResult.m_offset = fileMetadata.m_offset;
+                    listResult.m_crc32 = fileMetadata.m_crc32;
+
+                    listFileCallback(AZStd::move(listResult));
+                }
+                ++filePathIndex;
+            };
+        EnumerateFilePathIndexOffsets(EnumerateAllFiles, m_archiveToc.m_tocView);
+
+        // There are currently no error messages that enumerate file path sets
+        // So a default constructed instance which converts to boolean true is returned
+        return {};
+    }
+
+    bool ArchiveReader::DumpArchiveMetadata(AZ::IO::GenericStream& metadataStream,
+        const ArchiveMetadataSettings& metadataSettings) const
+    {
+        using MetadataString = AZStd::fixed_string<256>;
+        if (metadataSettings.m_writeFileCount)
+        {
+            auto fileCountString = MetadataString::format("Total File Count: %u\n", m_archiveHeader.m_fileCount);
+            metadataStream.Write(fileCountString.size(), fileCountString.data());
+        }
+
+        if (metadataSettings.m_writeFilePaths)
+        {
+            // Validate the file path and file metadata tables are in sync
+            if (m_archiveToc.m_tocView.m_filePathIndexTable.size() != m_archiveToc.m_tocView.m_fileMetadataTable.size())
+            {
+                auto errorString = MetadataString::format("Error: The Archive TOC of contents has a mismatched size between"
+                    " the file path index vector (size=%zu) and the file metadata vector (size=%zu).\n"
+                    "This indicates a code error in the ArchiveReader in keeping both vectors in sync and should never occur",
```

Comment:
```suggestion
                    "This indicates a code error in the ArchiveReader in keeping both vectors in sync and should never occur.",
```

Reply:
Also not sure if the "should never occur" should be here, its not something that is actionable outside of the ArchiveReader

Reply:
Yeah, I fix up the Error message to remove the should never occur comment.